lib/UsersStack.js

```diff
@@ -0,0 +1,154 @@
+        const oneUserResource = usersResource.addResource('{id}');
+
+        // customize 'bad request body' to show more helpful schema validation messages
+        new apigateway.CfnGatewayResponse(usersStack, 'apiBadRequestBody', {
+            restApiId: api.restApiId,
+            responseType: 'BAD_REQUEST_BODY',
+            statusCode: '400',
+            responseTemplates: {
+                'application/json': `{"message": $context.error.messageString, "errors": $context.error.validationErrorString}`,
+            },
+        });
+
+        // add validator for request body
+        const bodyValidator = new apigateway.RequestValidator(usersStack, 'bodyValidator', {
+            requestValidatorName: `${stackName}-bodyValidator`,
+            restApi: api,
+            validateRequestBody: true,
+            validateRequestParameters: false,
+        });
+
+        // create lambda layer
+        const crudLayer = new lambda.LayerVersion(usersStack, `${stackName}-CRUD-Layer`, {
+            layerVersionName: `${stackName}-CRUD-Layer`,
+            code: new lambda.AssetCode('src/layers/crud'),
+        });
+
+        // create lambdas for various CRUD actions
+        const crudActions = [
+            {
+                name: 'Create',
+                method: 'POST',
+                asset: 'src/lambdas/createUser/',
+                handler: 'createUser.handler',
+                resource: usersResource,
+                validator: bodyValidator,
+            },
+            {
+                name: 'Read',
+                method: 'GET',
+                asset: 'src/lambdas/readUser/',
+                handler: 'readUser.handler',
+                resource: oneUserResource,
+            },
+            {
+                name: 'Update',
+                method: 'PATCH',
+                asset: 'src/lambdas/updateUser/',
+                handler: 'updateUser.handler',
+                resource: oneUserResource,
+                validator: bodyValidator,
+            },
+            {
+                name: 'Delete',
+                method: 'DELETE',
+                asset: 'src/lambdas/deleteUser/',
+                handler: 'deleteUser.handler',
+                resource: oneUserResource,
+            },
+        ];
+
+        // map over CRUD actions
+        crudActions.map(entry => {
+            const {
+                name: crudName,
+                method: crudMethod,
+                asset: crudAsset,
+                handler: crudHandler,
+                resource: crudResource,
+                validator: crudValidator,
+            } = entry;
+
+            const crudLambda = new lambda.Function(usersStack, `${crudName}User`, {
+                functionName: `${stackName}-${crudName}User`,
+                code: new lambda.AssetCode(crudAsset),
+                handler: crudHandler,
+                runtime: lambda.Runtime.NODEJS_12_X,
+                memorySize: 1024,
```

Comment:
This could be fed in with environment variables, or computed in the CDK stack itself.  A variation could look to AWS SSM to avoid putting the values in the codebase.  Another option is to bundle config files with each lambda with definitions for each environment.  Let me know if you have any other techniques or interesting ideas here.